lib/compilers/scss-compiler.js

```diff
@@ -1,19 +1,55 @@
 const path = require('path')
 const fs = require('fs')
 const ensureRequire = require('../ensure-require')
-const cwd = process.cwd()
 const logger = require('../logger')
 
-const rewriteImports = (content, filePath) => content.replace(/@import\s+(?:'([^']+)'|"([^"]+)"|([^\s;]+))/g, (entire, single, double, unquoted) => {
-  const oldImportPath = single || double || unquoted
-  const absoluteImportPath = path.join(path.dirname(filePath), oldImportPath)
-  const lastCharacter = entire[entire.length - 1]
-  const quote = lastCharacter === "'" || lastCharacter === '"' ? lastCharacter : ''
-  const importPath = path.relative(cwd, absoluteImportPath)
-  return '@import ' + quote + importPath + quote
-})
+/**
+ * Resolves the path to the file locally.
+ *
+ * @param {String} to - the name of the file to resolve to
+ * @param {String} localPath - the local path
+ * @returns {String} path - path to the file to import
+ */
+function localResolve (to, localPath) {
+  if (localPath.startsWith('/')) {
+    return localPath
+  }
+
+  return path.join(path.dirname(to), localPath)
+}
 
-module.exports = (content, filePath, config) => {
+/**
+ * Applies the moduleNameMapper substitution from the jest config
+ *
+ * @param {String} source - the original string
+ * @param {String} filePath - the path of the current file (where the source originates)
+ * @param {Object} jestConfig - the jestConfig holding the moduleNameMapper settings
+ * @returns {String} path - the final path to import (including replacements via moduleNameMapper)
+ */
+function applyModuleNameMapper (source, filePath, jestConfig = {}) {
+  if (!jestConfig.moduleNameMapper) return source
+  const module = Array.isArray(jestConfig.moduleNameMapper) ? jestConfig.moduleNameMapper : Object.entries(jestConfig.moduleNameMapper)
```

Comment:
Thats a funny story actually. I had to introduce babel@7 to vue-jest locally while developing via `yarn link` to make it work properly with our current project (there seems to be some issue with yarn handling peerDependencies via `yarn link`). Anyway, with the new babel@7 setup, all of a sudden it returned an array instead of an object. I'll try to make a repro branch for this so we can look at the issue. I've also added a TODO and an explanation to the code in my (upcoming) fix.